File: apps/lastnotif/src/main/java/io/maru/lastnotif/LastNotifPollerAlarmReceiver.java
package io.maru.lastnotif;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Receives the 5-minute keepalive alarm and restarts the poller service
 * if it is not already running.
 */
public class LastNotifPollerAlarmReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        LastNotifStorage storage = new LastNotifStorage(context);
        if (storage.isServiceRunning()) {
            LastNotifPollerService.start(context);
            LastNotifPollerAlarmScheduler.schedule(context);
        }
    }
}

File: apps/lastnotif/src/main/java/io/maru/lastnotif/LastNotifPollerAlarmReceiver.kt
package io.maru.lastnotif

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.launch

class LastNotifPollerAlarmReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        val prefs = LastNotifPrefs(context)
        CoroutineScope(Dispatchers.IO).launch {
            if (prefs.serviceRunning.first()) {
                LastNotifPollerService.start(context)
                LastNotifPollerAlarmScheduler.schedule(context)
            }
        }
    }
}
